### ISSUE TITLE
Editing a record should recalculate closest dose

### ISSUE THREAD

**Issue by bannurowi** [CorruptedArk/did-i-take-my-meds#80]
**Describe the bug**
When editing a dose, it retains its connection to the original closest dose regardless of what dose is closest to the new time.

**To Reproduce**
1. Log a dose as taken.
2. Tap the dose record and edit the time taken.
3. The "closest dose" will remain the same, even if it is no longer actually the closest.

**Expected behavior**
Editing the taken time should recalculate which dose is the closest.

**Smartphone (please complete the following information):**
 - Device: Teracube 2e
 - OS: e/OS 1.12
 - Version: 1.6.6
 - Build Location: F-Droid release

**Additional context**
This is most relevant to me when I take a dose on time, but forget to log it in the app. By the time I remember, the next dose is assumed to be the "closest" one, so I can't log both doses as actually taken.

This was previously mentioned here: https://github.com/CorruptedArk/did-i-take-my-meds/issues/42#issuecomment-1179412265, but it's really a separate issue.

**Comment by nogweii** [comment]
Yeah, this would be very appreciated. In the mean time, you can export the database, edit the record, then restore the changed version.
